### PR TITLE
port the numba integration prototype from Enterprise to OSS, fixes feature-523

### PR DESCRIPTION
**Pull request by jmao-denver**
this prototype in its current design must be used in an unpublished, explicit way. So it should not break anything visible to users.

### CODE REVIEW

File: DB/src/main/java/io/deephaven/db/v2/sources/chunk/BooleanChunk.java [hunk 1]
```diff
@@ -14,11 +14,6 @@
 // region BufferImports
 // endregion BufferImports
 
-/**
- * {@link Chunk} implementation for boolean data.
- *
- * @IncludeAll
- */
```
Comment (devinrsmith) [inline DB/src/main/java/io/deephaven/db/v2/sources/chunk/BooleanChunk.java:21]: I don't think we should be getting rid of this javadoc - `@IncludeAll` is a special tag. I'm not sure if I explicitly deleted this in my original branch or not, but we should probably keep it. (This goes for the other chunk javadocs too.)

File: py/jpy/src/main/java/org/jpy/PyObject.java [hunk 2]
```diff
@@ -13,7 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  *
- * This file was modified by Deephaven Data Labs.
```
Comment (devinrsmith) [inline py/jpy/src/main/java/org/jpy/PyObject.java:16]: We shouldn't change this.

**Review by devinrsmith — CHANGES_REQUESTED** [review]
Looks good, mostly. Couple changes needed. Hard for me to tell if there are any big departures from how I originally structured my patch?

I'll be testing this out locally now.

**Comment by devinrsmith** [comment]
Great - was able to test the following, and it works:

```python
from deephaven import TableTools, java, numba as dnb

@dnb.vectorize
def my_manual(x: java.int, y: java.int) -> java.int:
  return (x % 3) + y

IJ = TableTools.emptyTable(1000000) \
    .view("I=(int)i", "J=(int)(i * 2)") \
    .update(java.formula(my_manual, name = "K", column_names = ["I", "J"]))
```

**Comment by devinrsmith** [comment]
Good example of the chunking working:

```python
from deephaven import TableTools, java, numba as dnb

@dnb.vectorize
def my_vectorized(x: java.int, y: java.int) -> java.int:
  return (x % 3) + y

vectorized_sum = TableTools.emptyTable(100000000) \
    .view("I=(int)i", "J=(int)(i * 2)") \
    .view(java.formula(my_manual, name = "K", column_names = ["I", "J"])) \
    .sumBy()
```

vs

```python
from deephaven import TableTools, java

def my_plain(x: java.int, y: java.int) -> java.int:
  return (x % 3) + y
  
plain_sum = TableTools.emptyTable(100000000) \
    .view("I=(int)i", "J=(int)(i * 2)") \
    .view("K = (int)my_plain(I, J)") \
    .sumBy()
```

File: DB/src/main/java/io/deephaven/db/v2/sources/chunk/BooleanChunk.java (hunk 1 above)
Comment (jmao-denver) [reply DB/src/main/java/io/deephaven/db/v2/sources/chunk/BooleanChunk.java:21]: the javadoc comments were most likely added after your commit. Will restore.

**Review by devinrsmith — APPROVED** [review]

**Pull request merged**